opm/grid/common/SubGridView.hpp

```diff
@@ -0,0 +1,371 @@
+// -*- tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*-
+// vi: set et ts=4 sw=2 sts=2:
+
+// Note: this file is based on defaultgridview.hh from Dune, and is therefore
+// licensed under the Dune license (GPLv2 + runtime exception),
+// see https://dune-project.org/about/license/
+// rather than the OPM license (GPLv3+)
+// Copyright 2021 Dune contributors.
+// Copyright 2021 SINTEF Digital, Mathematics and Cybernetics.
+
+#ifndef OPM_SUBGRIDVIEW_HEADER
+#define OPM_SUBGRIDVIEW_HEADER
+
+#include <dune/common/exceptions.hh>
+#include <dune/common/typetraits.hh>
+
+#include <dune/grid/common/capabilities.hh>
+#include <dune/grid/common/gridview.hh>
+
+#include <algorithm>
+#include <cassert>
+#include <cstddef>
+#include <map>
+#include <stdexcept>
+#include <type_traits>
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+
+namespace Dune
+{
+
+template <class GridImp>
+class SubGridView;
+
+template <class GridImp>
+struct SubGridViewTraits {
+    using GridViewImp = SubGridView<GridImp>;
+
+    /** \brief type of the grid */
+    using Grid = typename std::remove_const<GridImp>::type;
+
+    /** \brief type of the index set */
+    using IndexSet = typename Grid ::Traits ::LeafIndexSet;
+
+    /** \brief type of the intersection */
+    using Intersection = typename Grid ::Traits ::LeafIntersection;
+
+    /** \brief type of the intersection iterator */
+    using IntersectionIterator = typename Grid ::Traits ::LeafIntersectionIterator;
```

Comment:
Are we still accessing neighbors that are not in the subgrid?

Reply:
Yes.